frontend/utils.tsx

```diff
@@ -176,6 +176,15 @@ export function apiCheckPermission(
   return false
 }
 
+const chooseEndpoint = (input: [string, string] | string) => {
+  if (typeof input === 'string') {
+    return `/${input}/?format=json`
+  } else {
+    if (input[1].startsWith('/')) return input[1]
+    return chooseEndpoint(input[1])
+  }
```

Comment:
This seems like you can use an `else if` to make the code a bit clearer, and have the return statement at the end. Its good practice to usually put `{ ... }` even on one line statements, just to make the code easier to read and to reduce bugs if you expand the if statement.